rspace/src/main/scala/coop/rchain/rspace/history/KeySegment.scala

```diff
@@ -0,0 +1,54 @@
+package coop.rchain.rspace.history
+
+import scodec.bits.ByteVector
+
+import scala.annotation.tailrec
+
+final case class KeySegment(val value: ByteVector) {
```

Comment:
@nzpr mentioned that it can be implemented as [value class](https://docs.scala-lang.org/overviews/core/value-classes.html) to eliminate unnecessary wrappers and increase performance. Maybe it make sense?